### ISSUE TITLE
Feature Request: set code language for file coloring

### ISSUE THREAD

**Issue by cellocgw** [gitextensions/gitextensions#9572]
Requesting the capability to specify the source code language and have the file display's color-coding (comments, numbers, functions, text strings, etc)  adjusted accordingly.  Basically the same functionality as NotePad++ "Language" menu &plugin.    

If this is available, I apologize; couldn't find any mention of it in the documentation or Preferences dialog

**Comment by gerhardol** [comment]
The texteditor control supports highlighting when compiled only, so a dynamic setup will be quite big change.

The source files are available here:
https://github.com/gitextensions/ICSharpCode.TextEditor/tree/master/Project/Resources
More languages can be added.

**Comment by pmiossec** [comment]
Yes, if you want to add a new language or update one, you will have to compile your own version.

At the moment, the highlight is done based on the file extension.

I don't really understand if your request is to have a new or improve a language or being able to select one supported in a list.

Can you elaborate on your need? Want to add a new file support?

**Comment by cellocgw** [comment]
[responding to Philippe's post]  I was unable to find any mention in the help documentation or in the GUI itself of where I could specify the language for the file on display. 
WHile I may in the future design a source file for MATLAB, right now I mainly want to know how to specify the language to use when highlighting a file.

**Comment by gerhardol** [comment]
I have not seen any further guides for this than the code.
Try to use an existing language as an example.
